modules/react/button/lib/BaseButton.tsx

```diff
@@ -292,13 +133,295 @@ export const getPaddingStyles = (
+      paddingInlineEnd: cssVar(system.space.x8, space.l),
+    }),
+    largeEnd: cs({
+      paddingInlineStart: cssVar(system.space.x8, space.l),
+      paddingInlineEnd: cssVar(system.space.x6, space.m),
+    }),
+    mediumOnly: cs({padding: '0', minWidth: cssVar(system.space.x10, space.xl)}),
+    mediumStart: cs({
+      paddingInlineStart: cssVar(system.space.x5, '1.25rem'),
+      paddingInlineEnd: cssVar(system.space.x6, space.m),
+    }),
+    mediumEnd: cs({
+      paddingInlineStart: cssVar(system.space.x6, space.m),
+      paddingInlineEnd: cssVar(system.space.x5, '1.25rem'),
+    }),
+    smallOnly: cs({padding: '0', minWidth: cssVar(system.space.x8, space.l)}),
+    smallStart: cs({
+      paddingInlineStart: cssVar(system.space.x3, space.xs),
+      paddingInlineEnd: cssVar(system.space.x4, space.s),
+    }),
+    smallEnd: cs({
+      paddingInlineStart: cssVar(system.space.x4, space.s),
+      paddingInlineEnd: cssVar(system.space.x3, space.xs),
+    }),
+    extraSmallOnly: cs({padding: '0', minWidth: cssVar(system.space.x6, space.m)}),
+    extraSmallStart: cs({
+      paddingInlineStart: cssVar(system.space.x2, space.xxs),
+      paddingInlineEnd: cssVar(system.space.x3, space.xs),
+    }),
+    extraSmallEnd: cs({
+      paddingInlineStart: cssVar(system.space.x3, space.xs),
+      paddingInlineEnd: cssVar(system.space.x2, space.xxs),
+    }),
+  },
+});
+
+export function capitalize(string: string = '') {
+  return string.charAt(0).toUpperCase() + string.substring(1);
+}
+
+export function getIconPosition(
+  size?: keyof typeof buttonModifiers.size,
+  iconPosition?: IconPositions
+): keyof typeof buttonModifiers.iconPosition {
+  return `${size}${capitalize(iconPosition)}` as keyof typeof buttonModifiers.iconPosition;
+}
+
+/**
+ * The base button which all other buttons are built.
+ */
 export const BaseButton = createComponent('button')({
-  displayName: 'Button',
-  Component: ({children, ...elemProps}: ButtonContainerProps, ref, Element) => {
+  displayName: 'BaseButton',
+  Component: (
+    {
+      children,
+      cs,
+      size,
+      fillIcon,
+      iconPosition,
+      icon,
+      colors,
+      shouldMirrorIcon = false,
+      ...elemProps
+    }: ButtonContainerProps,
+    ref,
+    Element
+  ) => {
     return (
-      <ButtonContainer as={Element} ref={ref} type="button" {...elemProps}>
+      <Box
+        as={Element}
+        ref={ref}
+        type="button"
+        cs={[
+          baseButtonStyles,
+          cs,
+          buttonModifiers({
+            size: size,
+            iconPosition: getIconPosition(size, iconPosition),
+          }),
+          buttonVars.active(colors?.active || {}),
+          buttonVars.default(colors?.default || {}),
+          buttonVars.disabled(colors?.disabled || {}),
+          buttonVars.focus(colors?.focus || {}),
+          buttonVars.hover(colors?.hover || {}),
+        ]}
+        {...elemProps}
+      >
         {children}
-      </ButtonContainer>
```

Comment:
The `cs` prop should always go last. Otherwise if someone passes a `className`, all styling will be overridden.